### ISSUE TITLE
Crates.io and `cargo install zenith` 

### ISSUE THREAD

**Issue by rrichardson** [bvaisvil/zenith#5]
This is really slick!  I spend way too much of my time administering and monitoring machines.  I like the UX of this, it gives me the numbers I care about in an intuitive way. 

I would like to be able just `cargo install zenith` on machines.   

This means you'll need to work something out with the owner of https://crates.io/crates/zenith  or rename this.

**Comment by bvaisvil** [comment]
Thank you for the feedback! I'd love to publish on crates.io. Not sure the other project with that is under active development. Perhaps I'll reach out. Another issue is that I'm using a forked version of [sysinfo](https://github.com/GuillaumeGomez/sysinfo) that adds per process disk usage. I've got a pr open for [that](https://github.com/GuillaumeGomez/sysinfo/pull/248) but it hasn't yet been merged.

**Comment by karuppiah7890** [comment]
Dumb question: Should the repo name and crate name be the same? 🤔

**Comment by bvaisvil** [comment]
@karuppiah7890 That's a good point, I haven't really explored that.

I'll still need to either publish my own version of sysinfo into crates.io (don't like this idea), have my changes accepted or get the info from another library.

**Comment by apiraino** [comment]
I second this. I had read about `zenith` on a blog, executed `cargo install zenith` and then launched it. Got a weird output and came here to look for a bug report or more instructions.

I am dumb but it's too easy to install and launch the wrong binary (possibly executing something malicious).

**Comment by rrichardson** [comment]
The executable name doesn't have to match the crates.io name, but I think it has to match the Cargo.toml project name. 

Ripgrep is a good example.  `Cargo install ripgrep` gives you a binary called `rg` 

My recommendation would be to create a uniquely named project in crates.io.  Maybe `zenith-monitor` or something, but the binary can still be named `zenith`

**Comment by bvaisvil** [comment]
Immediate issue blocking progress on this is my git dependency on my own fork of the `sysinfo` crate.

**Comment by anahata0108** [comment]
Is there something special about your sysinfo fork that doesn't apply to the heim crate that you're also specifying a git repo for? I'm not understanding why this being a fork is a blocker.

**Comment by bvaisvil** [comment]
The `heim` crate is a git dependency because I needed a fix before an official release (still the case). I have made fixes and enhancements to `sysinfo` that I haven't yet made PRs for.

**Comment by anahata0108** [comment]
I understand the heim crate git dependency. I'm asking why you can't do the same with the sysinfo crate, since they're both on git. I must be missing something, I'm trying to find out what it is.

**Comment by bvaisvil** [comment]
For example, 
- currently `sysinfo` doesn't provide the niceness/priority of a pid, my fork does
- on linux `sysinfo`, the name of the process is truncated/missing, fixed on my fork
- on linux `sysinfo` was missing disks, fixed on my fork etc.

**Comment by anahata0108** [comment]
Right. I understand that. I think I'm not being clear. I'm sorry. What I mean is: what breaks with sysinfo if you publish zenith to crates.io as-is? What does having your own fork of sysinfo break when you do that? This is what I don't understand.

**Comment by bvaisvil** [comment]
You can't publish to `crates.io` [with git dependencies](https://github.com/rust-lang/cargo/issues/6738).

**Comment by anahata0108** [comment]
Ahh! Thank you! That explains everything.

**Comment by clouds56** [comment]
Is it still blocking after https://github.com/GuillaumeGomez/sysinfo/pull/287 was merged
